src/Microsoft.AspNet.OData.Shared/DefaultODataAPIHandler.cs

```diff
@@ -0,0 +1,157 @@
+        public override ODataAPIResponseStatus TryGet(IDictionary<string, object> keyValues, out TStructuralType originalObject, out string errorMessage)
+        {
+            ODataAPIResponseStatus status = ODataAPIResponseStatus.Success;
+            errorMessage = string.Empty;
+            originalObject = default(TStructuralType);
+
+            try
+            {
+                originalObject = GetFilteredItem(keyValues);
+
+                if (originalObject == null)
+                {
+                    status = ODataAPIResponseStatus.NotFound;
+                }
+            }
+            catch (Exception ex)
+            {
+                status = ODataAPIResponseStatus.Failure;
+                errorMessage = ex.Message;
+            }
+
+            return status;
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override ODataAPIResponseStatus TryCreate(IDictionary<string, object> keyValues, out TStructuralType createdObject, out string errorMessage)
+        {
+            createdObject = default(TStructuralType);
+            errorMessage = string.Empty;
+
+            try
+            {
+                if(originalList != null)
+                {
+                    originalList = new List<TStructuralType>();
+                }
+
+                createdObject = Activator.CreateInstance(_clrType) as TStructuralType;
+                originalList.Add(createdObject);
+
+                return ODataAPIResponseStatus.Success;
+            }
+            catch (Exception ex)
+            {
+                errorMessage = ex.Message;
+
+                return ODataAPIResponseStatus.Failure;
+            }
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        public override ODataAPIResponseStatus TryDelete(IDictionary<string, object> keyValues, out string errorMessage)
+        {
+            errorMessage = string.Empty;
+
+            try
+            {
+                TStructuralType originalObject = GetFilteredItem(keyValues);
+                originalList.Remove(originalObject);
+
+                return ODataAPIResponseStatus.Success;
+            }
+            catch (Exception ex)
+            {
+                errorMessage = ex.Message;
+
+                return ODataAPIResponseStatus.Failure;
+            }
+        }
+
+        public override IODataAPIHandler GetNestedHandler(TStructuralType parent, string navigationPropertyName)
+        {
+            foreach (PropertyInfo property in _clrType.GetProperties())
+            {
+                if (property.Name == navigationPropertyName)
+                {
+                    Type type = typeof(DefaultODataAPIHandler<>).MakeGenericType(property.PropertyType.GetGenericArguments()[0]);
```

Comment:
The assumption here is that the navigation property here will be generic. Is that assumption valid?

Reply:
If it's not generic, the customer can solve this by creating their custom `ODataAPIHandler<T>`